test/suite/Annotations.test.php

```diff
@@ -585,12 +590,149 @@ protected function testGetAnnotationsFromTrait()
             return;
         }
 
-        eval('trait TestTrait { }');
-        $this->setExpectedException(
-            self::ANNOTATION_EXCEPTION,
-            "Reading annotations from interface/trait 'TestTrait' is not supported"
-        );
-        Annotations::ofClass('TestTrait', '@note');
+        $annotations = Annotations::ofClass('SimpleTrait', '@note');
+
+        $this->check(count($annotations) === 1, 'SimpleTrait has one note annotation.');
+    }
+
+    protected function testCanGetMethodAnnotationsIncludedFromTrait()
+    {
+        if (version_compare(PHP_VERSION, '5.4.0', '<')) {
+            $this->pass();
+            return;
+        }
+
+        $annotations = Annotations::ofMethod(new \ReflectionClass('SimpleTraitTester'), 'runFromTrait');
+        $this->check(count($annotations) > 0, 'from class reflection and method name');
+
+        $annotations = Annotations::ofMethod(new \ReflectionMethod('SimpleTraitTester', 'runFromTrait'));
```

Comment:
This is redundant, because here you're not testing that trait annotations are found, but rather that `Annotations::ofMethod` accepts different type arguments (`ReflectionClass` and `ReflectionMethod`). This stuff should already be tested somewhere else I hope.